include/nbl/builtin/hlsl/sampling/concentric_mapping.hlsl

```diff
@@ -0,0 +1,46 @@
+#ifndef _NBL_BUILTIN_HLSL_SAMPLING_CONCENTRIC_MAPPING_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SAMPLING_CONCENTRIC_MAPPING_INCLUDED_
+
+#include "nbl/builtin/hlsl/tgmath.hlsl"
+#include "nbl/builtin/hlsl/math/functions.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+
+template<typename T>
+vector<T,2> concentricMapping(vector<T,2> _u)
+{
+    //map [0;1]^2 to [-1;1]^2
+    vector<T,2> u = 2.0f * _u - 1.0f;
+
+    vector<T,2> p;
+    #ifdef __HLSL_VERSION
+    if (nbl::hlsl::all<vector<bool,2> >(u == (vector<T,2>)(0.0)))
+    #else
+    if (u == (vector<T,2>)(0.0))
+    #endif
```

Comment:
this ifdef is awful, add `equals` to `nbl::hlsl::glsl::` in `glsl_compat` then you can do in both HLSL and C++
```cpp
if (all(glsl::equals(promote<vector<T,2> >(0.f),u)))
```

Reply:
I believe this is because `float3 == float3` in HLSL evaluates to `bool3` but in c++ it evaluates to a `bool`. For `glm` vectors use `equal`. There was also something preventing overriding `==` in c++.